src/pages/Main-screen.js

```diff
@@ -0,0 +1,72 @@
+import React, {useState, useEffect} from 'react';
+import {StyleSheet, View, ScrollView, RefreshControl} from 'react-native';
+import {NavBar} from '../../src/components/Nav-bar';
+import {SearchInput} from '../../src/components/Search-input';
+import {Card} from '../../src/components/Card';
+import {createNativeStackNavigator} from '@react-navigation/native-stack';
+
+const randomPick = {uri: 'https://picsum.photos/100'};
+const BASE_URL = 'https://demo.spreecommerce.org';
+const API_PATH = `${BASE_URL}/api/v2/storefront/products`;
+const Stack = createNativeStackNavigator();
+
+const wait = timeout => {
+  return new Promise(resolve => setTimeout(resolve, timeout));
+};
+
+export const MainScreen = () => {
+  const [backData, getBackData] = useState();
+
+  function fetchRequest() {
+    fetch(API_PATH)
+      .then(response => response.json())
+      .then(data => getBackData(data));
+  }
+
+  useEffect(() => {
+    fetchRequest();
+  }, []);
+
+  const RenderCards = backData?.data.map(item => (
+    <Card key={item.id} data={item.attributes} icon={randomPick} />
+  ));
+
+  const [refreshing, setRefreshing] = React.useState(false);
+
+  const onRefresh = React.useCallback(() => {
+    setRefreshing(true);
+    wait(2000).then(() => setRefreshing(false));
+    fetchRequest();
+  }, []);
+
+  return (
+    <>
+      <NavBar />
```

Comment:
As already said, I would propose to customize Header from React Navigation instead of using your own. While it is possible to use you own Header component, this should be done in some very specific case when default header is not suitable at all.